File: Clase 16/JAVA/ejemploAnimales/Perro.java
package ejemploAnimales;

public class Perro extends Animal{
    protected int poderMandibula;

    public Perro(int id, String nombre, String especie, int poderMandibula){
        super(id, nombre, especie);
        this.poderMandibula = poderMandibula;
    }

    public void haceGuau(){
        System.out.println(this.nombre + " hace GUAU!!!!");
    }
}

File: Clase 16/PYTHON/ejemplo_animales/Perro.py
from Animal import Animal

class Perro(Animal):

    def __init__(self, id, nombre, especie, poder_mandibula) -> None:
        super().__init__(id, nombre, especie)
        self.poder_mandibula = poder_mandibula
    
    def hace_guau(self):
        print(f"{self.nombre} hace GUAUUUU!!!!")